contracts/AngstromRouterAndHook.sol

```diff
@@ -18,8 +19,7 @@ import { SingletonAuthentication } from "@balancer-labs/v3-vault/contracts/Singl
 import { BatchRouterHooks } from "@balancer-labs/v3-vault/contracts/BatchRouterHooks.sol";
 import { RouterCommon } from "@balancer-labs/v3-vault/contracts/RouterCommon.sol";
 
-// TODO interface
-contract AngstromRouter is BatchRouterHooks, SingletonAuthentication {
+contract AngstromRouterAndHook is IBatchRouter, BatchRouterHooks, SingletonAuthentication {
     uint256 internal _lastUnlockBlockNumber;
 
     error OnlyOncePerBlock();
```

Comment:
Unsolicited review :) Consider naming the error `NotAngstromNode` vs. just `NotNode`.

Reply:
So, since there's already an implementation of this elsewhere, and I'm just porting, I thought it would be nice to keep the original namings (Angstrom team probably has a way to treat these reverts already, I didn't want to change what they're used to). The implementation is here: https://github.com/SorellaLabs/angstrom/blob/main/contracts/src/modules/UnlockHook.sol